pkg/clustermanager/eksa_mover.go

```diff
@@ -0,0 +1,146 @@
+		clientFactory:      clientFactory,
+		moveClusterTimeout: applyClusterSpecTimeout,
+		retryBackOff:       retryBackOff,
+	}
+
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	return m
+}
+
+// WithMoverNoTimeouts disables the timeout for all the waits and retries in management upgrader.
+func WithMoverNoTimeouts() MoverOpt {
+	return func(a *Mover) {
+		maxTime := time.Duration(math.MaxInt64)
+		a.moveClusterTimeout = maxTime
+	}
+}
+
+// WithMoverApplyClusterTimeout allows to configure how long the mover retries
+// to apply the objects in case of failure.
+// Generally only used in tests.
+func WithMoverApplyClusterTimeout(timeout time.Duration) MoverOpt {
+	return func(m *Mover) {
+		m.moveClusterTimeout = timeout
+	}
+}
+
+// WithMoverRetryBackOff allows to configure how long the mover waits between requests
+// to update the cluster spec objects and check the status of the Cluster.
+// Generally only used in tests.
+func WithMoverRetryBackOff(backOff time.Duration) MoverOpt {
+	return func(m *Mover) {
+		m.retryBackOff = backOff
+	}
+}
+
+// Move applies the cluster's namespace and spec without checking for reconcile conditions.
+func (m *Mover) Move(ctx context.Context, spec *cluster.Spec, fromCluster, toCluster *types.Cluster) error {
+	m.log.V(3).Info("Moving the cluster object")
+	err := retrier.New(
+		m.moveClusterTimeout,
+		retrier.WithRetryPolicy(retrier.BackOffPolicy(m.retryBackOff)),
+	).Retry(func() error {
+		var err error
+		fromClient, err := m.clientFactory.BuildClientFromKubeconfig(fromCluster.KubeconfigFile)
+		if err != nil {
+			return errors.Wrap(err, "building client to read cluster objects")
+		}
+
+		toClient, err := m.clientFactory.BuildClientFromKubeconfig(toCluster.KubeconfigFile)
+		if err != nil {
+			return errors.Wrap(err, "building client to move cluster objects")
+		}
+
+		// read the cluster from bootstrap
+		cluster := &v1alpha1.Cluster{}
+		if err := fromClient.Get(ctx, spec.Cluster.Name, spec.Cluster.Namespace, cluster); err != nil {
+			return errors.Wrapf(err, "getting cluster from %s", fromCluster.Name)
+		}
+
+		// pause cluster on bootstrap
+		cluster.PauseReconcile()
+		if err := fromClient.Update(ctx, cluster); err != nil {
+			return errors.Wrapf(err, "updating cluster on %s", fromCluster.Name)
+		}
+
+		if err := moveClusterResource(ctx, cluster, toClient); err != nil {
+			return err
+		}
+
+		if err := moveChildObjects(ctx, *spec, fromClient, toClient); err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	return err
+}
+
+func moveClusterResource(ctx context.Context, cluster *v1alpha1.Cluster, client kubernetes.Client) error {
+	cluster.ResourceVersion = ""
+	cluster.UID = ""
+
+	// move eksa cluster
+	if err := client.Create(ctx, cluster); err != nil && !apierrors.IsAlreadyExists(err) {
+		return errors.Wrapf(err, "moving cluster %s", cluster.Name)
+	}
+
+	return nil
+}
+
+func moveChildObjects(ctx context.Context, spec cluster.Spec, fromClient, toClient kubernetes.Client) error {
```

Comment:
why pass the `cluster.Spec` instead of a pointer? Curious what we are gaining there

Reply:
We should pass the pointer. I removed the pointer when testing if range allocated new memory or not, but now it's back.